system_tests/datastore.py

```diff
@@ -271,7 +287,13 @@ def test_projection_query(self):
         self.assertEqual(catelyn_stark_dict,
                          {'name': 'Catelyn', 'family': 'Stark'})
 
-        catelyn_tully_entity = entities[3]
+        if EMULATOR_DATASET is None:
```

Comment:
When answering this query, the ordering of `name` and `family` don't matter. My guess is the emulator is using the preferred index (lexographical) (family, name) where production is using an index that already exists (name, family).

This fits with the chosen indexes:

name      |  family     
--------- | ------------
Arya      |  Stark      
Bran      |  Stark      
Eddard    |  Stark      
Catelyn   |  Stark      
Catelyn   |  Tully      
Jon Snow  |  Stark      
Rickard   |  Stark    
Robb      |  Stark      
Sansa     |  Stark      

family  | name      
------- | --------
Stark   | Arya      
Stark   | Bran      
Stark   | Eddard    
Stark   | Catelyn   
Stark   | Jon Snow  
Stark   | Rickard   
Stark   | Robb      
Stark   | Sansa     
Tully   | Catelyn   

For the test I would definitely force an order:

    SELECT name, family from Characters WHERE __key__ HAS ANCESTOR(Key('Book', 'GoT')) ORDRE BY name, family;